platform/default/mbgl/storage/offline_database.cpp

```diff
@@ -458,6 +466,7 @@ void OfflineDatabase::deleteRegion(OfflineRegion&& region) {
     stmt->run();
 
     evict(0);
+    db->exec("PRAGMA incremental_vacuum");
```

Comment:
Note that `PRAGMA incremental_vacuum` is not the same as `VACUUM`. In particular, my interpretation of the documentation is that `PRAGMA incremental_vacuum` is much cheaper: it works with a page structure that has already been prepared for incremental vacuum via `PRAGMA auto_vacuum`, and it doesn't compact partially filled pages of the database nor rewrite the entire database file like `VACUUM` does. I am not sure that a single `PRAGMA incremental_vacuum` after deleting several regions would be significantly cheaper than calling `PRAGMA incremental_vacuum` after each deletion. In fact, it might block other database commands for a longer continuous period of time.

Regarding the second point, we expect there the number of free pages for the ambient cache to be low. The expected behavior is that the ambient cache fills up to the maximum allotted size, and then stays at that size. Vacuuming free pages from it would in fact be a deoptimization, because those pages would just need to be reallocated as soon as the cache filled back up to its maximum size.

I'd prefer to keep the vacuum strategy a private implementation detail for now. There are certainly some unknowns about how it will perform, but it's not clear to me that providing API control will help.